### PR TITLE
feat: add router for contribution fees

### PR DESCRIPTION
**Pull request by 0xble**
Implements a crowdfund router that allows contributions to include a flat fee before being passed on to the crowdfund

### CODE REVIEW

**Comment by arr00** [comment]
We shouldn't use globals here for who can call the claim fees fn. Let's pass an owner on construction and have it be more generic wording.

**Comment by arr00** [comment]
For simplicity, we don't actually need to know which type of crowdfund the user is contributing in this contract. It can just be a low level eth call with the given calldata to the given crowdfund. That may be a better approach.

File: contracts/crowdfund/ContributionRouter.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    /// @notice Claim fees from the contract as the DAO.
+    /// @param recipient The address to send the fees to.
+    function claimFees(address payable recipient) external onlyPartyDao {
+        recipient.transferEth(address(this).balance);
+
+        emit ClaimedFees(msg.sender, recipient);
+    }
+
+    receive() external payable {}
```
Comment (arr00) [inline contracts/crowdfund/ContributionRouter.sol:90]: Why do we need this?

**Comment by 0xble** [comment]
> For simplicity, we don't actually need to know which type of crowdfund the user is contributing in this contract. It can just be a low level eth call with the given calldata to the given crowdfund. That may be a better approach.

The ABIs are different. For example, `InitialETHCrowdfund.contribute()` has an extra `tokenId` param that `Crowdfund.contribute()` doesn't.

**Comment by arr00** [comment]
Do we care to be able to switch the owner of the wrapper? I assume its fine like this.

**Comment by 0xble** [comment]
> Do we care to be able to switch the owner of the wrapper? I assume its fine like this.

Since owner is going to be the DAO's multisig, it's unlikely to change. Worse case and it does change, can always deploy a new one with an updated owner and switch over to that.

**Pull request merged**